### ISSUE TITLE
std::list leaks memory while std::vector does not

### ISSUE THREAD

**Issue by saraedum** [wlav/cppyy#245]
The following quickly eats through my RAM:

```python
import cppyy
cppyy.cppdef(r"""
#include <list>
std::list<int> f() { return std::list<int>(3); }
""")
while True: _ = list(cppyy.gbl.f())
```

However, if I replace `list` with `vector`, memory consumption is stable.

---

This is with cppyy 3.1.2 from conda-forge on x86_64 Arch Linux.

**Comment by saraedum** [comment]
Same with a pip installed cppyy 3.1.2.

**Comment by wlav** [comment]
Fixed in repo (and leak check test added). The reason for the difference between `list` and `vector` is that `std::vector` has its own ("high performance") iterator. The cause was an incorrect lifeline direction (causing a circular dependency) in the generic STL iterator protocol, which was unnecessary anyway b/c the iterator pythonization already added a lifeline.

**Comment by wlav** [comment]
Fix released with cppyy 3.5.0 and its dependencies.
